spring-integration-file/src/main/java/org/springframework/integration/file/dsl/FileTransferringMessageHandlerSpec.java

```diff
@@ -52,6 +52,9 @@
 
 	private DefaultFileNameGenerator defaultFileNameGenerator;
 
+	protected FileTransferringMessageHandlerSpec() {
```

Comment:
I would consider extra generic argument on this class like:
```
FileTransferringMessageHandlerSpec<F, H extends FileTransferringMessageHandler<F>, S extends FileTransferringMessageHandlerSpec<F, H, S>>
		extends MessageHandlerSpec<S, H>
```

So, all the implementations would follow requirements.
But this indeed going to be a breaking change.
So, let's leave your empty ctor solution for time being with `//TODO` to fix in the next version via generics!

Reply:
I'll add the TODO

Reply:
it's ok now